### PR TITLE
Backport #2859 to release/2.1.x

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Backport #2859 to release/2.1.x

### CODE REVIEW

**Comment by antonfirsov** [comment]
Please squash this mess if it will ever build 😅

I've seen some unrelated WEBP test failures locally, probably they will show up here.

**Comment by antonfirsov** [comment]
Even BMP failures. I don't understand this, neither we changed the product code nor we added tests since 2.1.9. Will continue investigating tomorrow, but if anyone has an idea, let me know!

**Comment by JimBobSquarePants** [comment]
@antonfirsov Github keeps changing what’s installed on images. I discovered with V3 branch that weirdly you had to install NET 8 along with the current target version for anything to build. I would use the yaml file from that branch as the basis and change the target frameworks.

**Comment by brianpopow** [comment]
I cannot reproduce the bmp test failures on windows. When I run:
```
dotnet test -c Release --filter "Format=Bmp"
```

All tests are green.

File: tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs [hunk 1]
```diff
@@ -611,6 +611,11 @@ public void BmpDecoder_CanDecode_Os2v2Header<TPixel>(TestImageProvider<TPixel> p
         public void BmpDecoder_CanDecode_Os2BitmapArray<TPixel>(TestImageProvider<TPixel> provider)
             where TPixel : unmanaged, IPixel<TPixel>
         {
+            if (TestEnvironment.IsLinux)
+            {
+                throw new SkipTestException("See discussion on https://github.com/SixLabors/ImageSharp/pull/2890");
+            }
```
Comment (antonfirsov) [inline tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs:617]: The Linux failure is a mystery: it only shows when running the test suite. If I run a console app against the compiled `ImageSharp.dll` on Ubuntu, the image loads without an issue. I have spent an hour trying to hook a debugger into VS Code, but it keeps ignoring my breakpoints.

I ended up skipping the test on Linux. (1) It's likely a false alert that appears only with `dotnet test` (2) likely not a regression against the previous version, (3) an edge-case anyways.

Comment (brianpopow) [reply tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs:617]: I think it is ok to skip those bmp tests on linux, they are extremely niche bitmap variants. It's also a mystery to me why this is different on linux than on windows.

Comment (antonfirsov) [reply tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs]: It was an LFS issue (╯°□°)╯︵ ┻━┻

Comment (brianpopow) [reply tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs]: Wow, good find

**Comment by antonfirsov** [comment]
@JimBobSquarePants it should be good to merge this now and start publishing stuff.

**Review by JimBobSquarePants — APPROVED** [review]
LGTM Thanks! 👍
